src/many-modules/src/_9_account/features/multisig.rs

```diff
@@ -12,9 +12,48 @@ use many_types::ledger::TokenAmount;
 use many_types::Timestamp;
 use minicbor::bytes::ByteVec;
 use minicbor::data::Type;
-use minicbor::{encode, Decode, Decoder, Encode, Encoder};
+use minicbor::{encode, decode, Decode, Decoder, Encode, Encoder};
 use std::collections::{BTreeMap, BTreeSet};
 
+/// A short note in a transaction
+#[derive(Clone, Debug, PartialEq)]
+// Using AsRef<str> for extensibility:
+// We might want to borrow a &Memo<str> instead of a &Memo<String>
+pub struct Memo<S: AsRef<str>>(S);
+
+impl<S, C> Encode<C> for Memo<S> where S: AsRef<str> {
+    fn encode<W: encode::Write>(&self, e: &mut Encoder<W>, _: &mut C) -> Result<(), encode::Error<W::Error>> {
+        e.str(self.0.as_ref()).map(|_| ())
+    }
+}
+
+impl<C> Decode<'_, C> for Memo<String> {
+    fn decode(d: &mut Decoder<'_>, _: &mut C) -> Result<Self, decode::Error> {
+        let s = d.str()?;
+        if s.len() > MULTISIG_MEMO_DATA_MAX_SIZE {
```

Comment:
So a bug in the previous implementation that I let slide was that it's possible to create a Memo with a longer length and encode it. It would be even better if the construction checked for the size, using maybe TryInto instead of into. What do you think?

Reply:
I think it's a great idea

Reply:
@hansl addressed